### ISSUE TITLE
Uninstalling the core add-on doesn't uninstall study add-ons and causes confusing UX

### ISSUE THREAD

**Issue by jonathanmayer** [mozilla-rally/rally-core-addon#635]
If you uninstall the Rally core add-on, there's a data deletion ping to the Telemetry backend, but study add-ons remain installed. That might confuse users, especially since the Rally support library continues to automatically open the "Hold up. You're missing the Rally add-on!" page on browser startup. A possible fix would be for the Rally support library to add a `browser.management.onUninstalled` listener, and if the core add-on is uninstalled, treat that as if the core add-on had sent an uninstall message to the study add-on.

https://github.com/mozilla-rally/rally-core-addon/blob/447444216b7bc50e0239339f365325cf93b711ea/support/rally.js#L166

**Comment by hamilton** [comment]
@jonathanmayer when you say "uninstall the Rally core add-on" do you mean through the core add-on options page or manually? In the former case, it _should_ uninstall all studies.

**Comment by jonathanmayer** [comment]
Manually uninstalling on `about:addons`. Leaving Rally from the options page works fine.

**Comment by rhelmer** [comment]
> If you uninstall the Rally core add-on, there's a data deletion ping to the Telemetry backend, but study add-ons remain installed. That might confuse users, especially since the Rally support library continues to automatically open the "Hold up. You're missing the Rally add-on!" page on browser startup. A possible fix would be for the Rally support library to add a `browser.management.onUninstalled` listener, and if the core add-on is uninstalled, treat that as if the core add-on had sent an uninstall message to the study add-on.
> 
> https://github.com/mozilla-rally/rally-core-addon/blob/447444216b7bc50e0239339f365325cf93b711ea/support/rally.js#L166

Yes I believe this would work, but it would require studies to [have the `management` permission](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/management/onUninstalled). I don't think that's a deal-breaker necessarily, we just try to use the minimal permissions necessary.

**Comment by jonathanmayer** [comment]
A couple other approaches come to mind.
* Poll install state with messages between add-ons. For example, on startup, the Rally support library could send a message to the core add-on. If the core add-on doesn't respond within some period of time (maybe with a little retrying to account for add-on startup sequencing and upgrading), assume the core add-on is gone and uninstall.
* Push install state with a `Port` between add-ons. For example, on startup, the Rally support library could open a `Port` to the core add-on. If the `Port` errors out, that's an indication that the core add-on might be gone.
